### DISCUSSION TITLE
UseRequestContext decorator order

### DISCUSSION THREAD

**Discussion by lucasgbsampaio-me** [mikro-orm/nestjs#148] [Q&A] [answered]
Why `@UseRequestContext` should be used in another method if its combined with other decorators ? Does the order of appearance not apply?

**Comment by B4nan** [comment]
The method always creates new context, that's why you shouldn't nest those, its not about order.

Or what do you mean, other decorators? You can combine it anyway you want, its just a common source of bugs and having it separately makes things clear.

**Reply by lucasgbsampaio-me** [reply]
@B4nan if i implement this code, won't it create a new context every time a message in the queue is processed?

````
  @Process(Events.Validate)
  @UseRequestContext()
  async process(job: Job) {
    await this.repository.findById('1');
  }
````

**Reply by B4nan** [reply, marked as answer]
Yeah it should, but I think there were some cases where the order of decorators mattered. I don't have any examples, if it works for you, it works for you :] (and if it doesn't you would get a validation error for using the global context)
